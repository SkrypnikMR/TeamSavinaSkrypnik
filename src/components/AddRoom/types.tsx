import React from 'react';

export interface IDiv {
    background: string;
    bottom: string;
    right: string;
    padding: string;
    fontSize: string;
    fontWeight: string;
    width: string;
    height: string;
    color: string;
    borderRadius: string;
    lineHeight: string;
    textAlign: string;
    margin: string;
    cursor: string;
    webkitTransform: string;
    msTransform: string;
    transform: string;
    boxShadow: string;
    transitionDuration: string;
}
