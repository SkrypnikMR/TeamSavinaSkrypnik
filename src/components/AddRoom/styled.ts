import styled from 'styled-components';

import { IDiv } from './types';
import { midGrey, bgColorDefault } from '../UI/baseLayout';

export const StAddRoom = styled.div < IDiv > `
    background: ${midGrey};
    bottom: 0px;
    right: 0px;
    padding: 2px;
    font-size: 30px;
    font-weight: bold;
    width: 30px;
    height: 30px;
    color: ${bgColorDefault};
    border-radius: 50%;
    line-height: 33px;
    text-align: center;
    margin: 0 10px 30px;
    cursor: pointer;
    :hover {
        -webkit-transform: scale(1.005);
        -ms-transform: scale(1.005);
        transform: scale(1.005);
        box-shadow: 0 14px 28px rgba(0, 0, 0, 0.6), 0 10px 10px rgba(0, 0, 0, 0.22);
        transition-duration: 0.3s;
    }
`;
