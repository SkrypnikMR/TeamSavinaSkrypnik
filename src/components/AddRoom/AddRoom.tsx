import React from 'react';
import { StAddRoom } from './styled';

const AddRoom = () => {
    return (
        <StAddRoom>
            +
        </StAddRoom>
    );
};

export default AddRoom;
