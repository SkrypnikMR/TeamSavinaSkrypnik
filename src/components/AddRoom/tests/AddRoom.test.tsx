import React from 'react';
import ReactDOM from 'react-dom';
import configureStore from 'redux-mock-store';

import { shallow, mount } from 'enzyme';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';

import AddRoom from '../AddRoom';

import { act } from 'react-dom/test-utils';

const element = document.createElement('div');
element.setAttribute('id', 'modal');
element.setAttribute('data-testid', 'modal-test-id');

jest
    .spyOn(ReactDOM, 'createPortal')
    .mockImplementation((children, c, key) => {
        const symbol = Symbol.for('react.portal');
        return {
            $$typeof: symbol,
            key: key == null ? null : '' + key,
            children,
            containerInfo: element,
            implementation: null,
            type: symbol.description,
            props: null,
        } as ReactPortal;
    });

const mockStore = configureStore();
const store = mockStore({
  game: {
    rooms: [],
    // userLogin: 'login',
  },
});

describe('AddRoom', () => {
    it('Should match snapshot', () => {
        const component = shallow(<AddRoom />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StAddRoom', () => {
        const component = mount(<AddRoom/>);
        expect(component.find('styled__StAddRoom')).toHaveLength(1);
    });
    it('Should', () => {
        const component = mountSmart(<AddRoom />, store);
        console.log(component.debug());
        component.find('styled__StAddRoom').simulate('click');
        expect(component.find('ModalCreateRoom')).toHaveLength(1);
    });
});
