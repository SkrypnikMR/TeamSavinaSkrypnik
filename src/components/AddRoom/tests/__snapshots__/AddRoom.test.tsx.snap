// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoom Should match snapshot 1`] = `"<div class=\\"styled__StAddRoom-sc-5w938c-0 djgSFI\\">+</div>"`;
